### PR TITLE
Add a convention to expose logging level.

### PR DESCRIPTION
**Pull request by jessemyers**
Next up: allow runtime changes

Example (subset of a large) output :

```
{
  "children": [
    {
      "children": [], 
      "level": "INFO", 
      "name": "SNSProducer"
    }, 
    {
      "children": [], 
      "level": "INFO", 
      "name": "SQSMessageDispatcher"
    }, 
    {
      "children": [], 
      "level": "INFO", 
      "name": "SQSMessageHandlerRegistry"
    }, 
    {
      "children": [], 
      "level": "INFO", 
      "name": "SleepPolicy"
    },
    {
      "children": [
        {
          "children": [], 
          "level": "WARNING", 
          "name": "swagger_spec_validator.ref_validators"
        }, 
        {
          "children": [], 
          "level": "WARNING", 
          "name": "swagger_spec_validator.util"
        }, 
        {
          "children": [], 
          "level": "WARNING", 
          "name": "swagger_spec_validator.validator12"
        }, 
        {
          "children": [], 
          "level": "WARNING", 
          "name": "swagger_spec_validator.validator20"
        }
      ], 
      "level": "WARNING", 
      "name": "swagger_spec_validator"
    }, 
    {
      "children": [], 
      "level": "INFO", 
      "name": "werkzeug"
    }
  ], 
  "level": "INFO", 
  "name": ""
}

### CODE REVIEW

**Comment by netama** [comment]
LGTM

File: microcosm_flask/conventions/logging_level.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+"""
```
Comment (cdm-ium) [inline microcosm_flask/conventions/logging_level.py]: Is it too meta to test?

File: microcosm_flask/conventions/logging_level.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    name = fields.String(required=True)
+    level = fields.String(required=True)
+    children = fields.List(fields.Nested("LoggerSchema"), required=True)
+
+
+class LoggingLevel(object):
+    pass
+
+
+class LoggingLevelConvention(Convention):
```
Comment (cdm-ium) [inline microcosm_flask/conventions/logging_level.py]: In practice, it's very easy to forget to unset logging level changes and blow through loggly quotas. Do you think we could build a timeout into the convention?

File: microcosm_flask/conventions/logging_level.py (hunk 1 above)
Comment (jessemyers) [inline microcosm_flask/conventions/logging_level.py]: Nope, that'll come; just wanted to see if anyone objected to the structure first before I entrenched it with tests.

File: microcosm_flask/conventions/logging_level.py (hunk 2 above)
Comment (jessemyers) [inline microcosm_flask/conventions/logging_level.py]: I'm not sure if that's possible. I'll look.

Comment (jessemyers) [inline microcosm_flask/conventions/logging_level.py]: Answer: it's possible but it's a bit of a hack.

 - We have to write a class that wraps an integer (logging level number)
 - We have to override comparison operations to use a timeout

Worth while?

Comment (wheremiah) [inline microcosm_flask/conventions/logging_level.py]: 🕦  Late reply... but I think this is a pretty nice idea 👍

**Pull request closed without merging**